src/dev/flang/be/interpreter2/JavaInterface.java

```diff
@@ -0,0 +1,628 @@
+        result = null;
+      }
+    catch (ClassNotFoundException e)
+      {
+        Errors.fatal("ClassNotFoundException when calling fuzion.java.get_static_field for field "+clazz+"."+field);
+        result = null;
+      }
+    catch (NoSuchFieldException e)
+      {
+        Errors.fatal("NoSuchFieldException when calling fuzion.java.get_static_field for field "+clazz+"."+field);
+        result = null;
+      }
+    return result;
+  }
+
+
+  /**
+   * forName load a class with a given name using this accessible
+   * object's class loader.
+   *
+   * @param name the class name using "." as separator between package
+   * and class name.
+   *
+   * @return the loaded class.
+   */
+  static Class forName(String name)
+  {
+    Class result;
+    try
+      {
+        result = Class.forName(name);
+      }
+    catch (ClassNotFoundException e)
+      {
+        result = null;
+      }
+    return result;
+  }
+
+
+
+  /**
+   * str2type converts a type descriptor of a field into the corresponding type.
+   *
+   * @param str a type descriptor, e.g. "Z", "Ljava/lang/String;".
+   *
+   * @return the type, e.g. Boolean.TYPE, String.class, etc.
+   */
+  static Class str2type(String str) {
+    switch (str.charAt(0)) {
+    case 'Z': return Boolean.TYPE;
+    case 'B': return Byte.TYPE;
+    case 'C': return Character.TYPE;
+    case 'S': return Short.TYPE;
+    case 'I': return Integer.TYPE;
+    case 'J': return Long.TYPE;
+    case 'F': return Float.TYPE;
+    case 'D': return Double.TYPE;
+    case 'V': return Void.TYPE;
+    case '[': return forName(str                            .replace('/','.'));
+    case 'L': return forName(str.substring(1,str.length()-1).replace('/','.'));
+    }
+    return null;
+  }
+
+
+  /**
+   * getEnd find the end of a type string starting at index i of d.
+   *
+   * @param d the descriptor string
+   *
+   * @param i the current index
+   *
+   * @return the index after the last index of the subtype.
+   */
+  static int getEnd(String d, int i) { // end of a sub-type in descriptor
+    while (d.charAt(i) == '[') {
+      i++;
+    }
+    if (d.charAt(i) == 'L') {
+      return d.indexOf(';',i)+1;
+    } else {
+      return i+1;
+    }
+  }
+
+
+  /**
+   * getPars determines an array of parameter types of a given
+   * signature.
+   *
+   * @return a new parameter type array.
+   */
+  public static Class[] getPars(String d)
+  {
+    Class[] result;
+
+    /* count parameters: */
+    int cnt = 0;
+    int i = 1;
+    while (d.charAt(i)!=')')
+      {
+        int e = getEnd(d,i);
+        if (e <= i)
+          {
+            return null;
+          }
+        cnt++;
+        i = e;
+      }
+
+    result = new Class[cnt];
+
+    /* get parameters: */
+    cnt = 0;
+    i = 1;
+    while (d.charAt(i)!=')')
+      {
+        int e = getEnd(d,i);
+        result[cnt] = str2type(d.substring(i,e));
+        cnt++;
+        i = e;
+      }
+    return result;
+  }
```

Comment:
These apparently have been moved to `dev.flang.util.JavaInterface`, should not be copied here.